pkg/devicetemplateconverter/grpc.go

```diff
@@ -39,25 +43,30 @@ func (s *endDeviceTemplateConverterServer) ListFormats(ctx context.Context, _ *p
 }
 
 // Convert implements ttnpb.DeviceTemplateServiceServer.
-func (s *endDeviceTemplateConverterServer) Convert(req *ttnpb.ConvertEndDeviceTemplateRequest, res ttnpb.EndDeviceTemplateConverter_ConvertServer) error {
+func (s *endDeviceTemplateConverterServer) Convert(
+	req *ttnpb.ConvertEndDeviceTemplateRequest,
+	res ttnpb.EndDeviceTemplateConverter_ConvertServer,
+) error {
 	converter, ok := s.DTC.converters[req.FormatId]
 	if !ok {
 		return errNotFound.WithAttributes("id", req.FormatId)
 	}
 	ctx, cancel := errorcontext.New(res.Context())
+	ctx = devicetemplates.NewContextWithProfileIDs(ctx, req.GetEndDeviceVersionIds())
 	ch := make(chan *ttnpb.EndDeviceTemplate)
 	go func() {
 		if err := converter.Convert(ctx, bytes.NewReader(req.Data), ch); err != nil {
 			cancel(err)
 		}
 	}()
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case tmpl, ok := <-ch:
 			if !ok {
-				return nil
+				return ctx.Err()
```

Comment:
An extra observation for reviewers is that I changed this due to some testing locally where when converting one device what happened was that the channel was closed and there could be the scenario where the `ctx.Error` is set but the `case tmpl, ok := <-ch:` was chosen first and it being closed returned a non error when its supposed to do the opposite. 

`Select` might chose randomly if both cases happen at once. Considering that the `ctx.Err()` is `nil` if not set I think this is an appropriate solution.